### ISSUE TITLE
macOS support

### ISSUE THREAD

**Issue by talal** [majewsky/gofu#3]
I was having some problems on the `make install` part, the `make` worked fine:

but the `make install` gave the following error:
```
$ make install
GOPATH=/Users/talal/Go/src/github.com/majewsky/gofu/.gopath GOBIN=/Users/talal/Go/src/github.com/majewsky/gofu/build go install  -ldflags '-s -w' 'github.com/majewsky/gofu'
install -D -m 0755 build/gofu "/usr/bin/gofu"
install: illegal option -- D
usage: install [-bCcpSsv] [-B suffix] [-f flags] [-g group] [-m mode]
               [-o owner] file1 file2
       install [-bCcpSsv] [-B suffix] [-f flags] [-g group] [-m mode]
               [-o owner] file1 ... fileN directory
       install -d [-v] [-g group] [-m mode] [-o owner] directory ...
```

I removed the `-D` option from the Makefile.

then:
```
$ make install
GOPATH=/Users/talal/Go/src/github.com/majewsky/gofu/.gopath GOBIN=/Users/talal/Go/src/github.com/majewsky/gofu/build go install  -ldflags '-s -w' 'github.com/majewsky/gofu'
install -m 0755 build/gofu "/usr/bin/gofu"
install: /usr/bin/gofu: Operation not permitted
make: *** [install] Error 71
```

Usually user stuff/binaries go in /usr/local/bin, that is where homebrew also symlinks its installed binaries. So I changed the line 17-19 in the Makefile to:
```
install: FORCE all
	install -m 0755 build/gofu "/usr/local/bin/gofu"
	for APPLET in $(APPLETS); do ln -s gofu "/usr/local/bin/$${APPLET}"; done
```

that worked and the correct symlinks were made.

Since I have zero clue about how Makefile(s) work, kindly advise on what would be a better approach to make this work on macOS.

**Comment by majewsky** [comment]
Does the following command work for you?
```
install -d -m 0755 /usr/local/bin
```

**Comment by talal** [comment]
the flag error is gone but I get this:

```
$ make install
GOPATH=/Users/talal/x/src/github.com/majewsky/gofu/.gopath GOBIN=/Users/talal/x/src/github.com/majewsky/gofu/build go install  -ldflags '-s -w' 'github.com/majewsky/gofu'
install -d -m 0755 "/usr/local/bin"
for APPLET in i3status rtree prettyprompt; do ln -s gofu "/usr/bin/${APPLET}"; done
ln: /usr/bin/i3status: Operation not permitted
ln: /usr/bin/rtree: Operation not permitted
ln: /usr/bin/prettyprompt: Operation not permitted
make: *** [install] Error 1
```

Changing the install force section to the following works:
```
install: FORCE all
	install -d -m 0755 "/usr/local/bin"
	for APPLET in $(APPLETS); do ln -s gofu "/usr/local/bin/$${APPLET}"; done
```

**Comment by talal** [comment]
Edit: after a quick `make install --help`, I found that on macOS the directory flag is `-C` 🤦‍♂️ 

So changing the flag to `-C` and the `PREFIX` at the top to `/usr/local` works, no additional changes required.

This change also passes the `git` to `bit` test  :double facepalm:

**Comment by majewsky** [comment]
`install` as in `install -D -m 0755 ...` is not the same as `install` in `make install`.

`install` the command is `/usr/bin/install` ([manpage for the GNU version](https://linux.die.net/man/1/install)). `install` in a `make` invocation is a Makefile target.

I committed a fix; it should work for you with commit 5d966f1 if you do `make install PREFIX=/usr/local`.

**Comment by talal** [comment]
Thank you, works as intended 😄
